client/lib/upgrades/actions/domain-management.js

```diff
@@ -48,13 +47,7 @@ function setPrimaryDomain( siteId, domainName, onComplete = noop ) {
 			return onComplete( error, data );
 		}
 
-		sites.setSelectedSite( siteId );
-
-		Dispatcher.handleServerAction( {
-			type: 'FETCH_SITES'
-		} );
-
-		sites.once( 'change', () => {
+		requestSite( siteId )( dispatch ).then( () => {
```

Comment:
It's more complicated than that. Somehow sites and domains depend on each other because of the url structure we use. It's better to update them at the same time to avoid unexpected race conditions. I don't know the details, but there is room for improvement here :)

I mirrored existing logic to make it work.